clippy_lints/src/use_self.rs

```diff
@@ -56,10 +56,16 @@ impl LintPass for UseSelf {
 const SEGMENTS_MSG: &str = "segments should be composed of at least 1 element";
 
 fn span_use_self_lint(cx: &LateContext<'_, '_>, path: &Path) {
+    // path segments only include actual path, no methods or fields
+    let last_path_span = path.segments.last().expect(SEGMENTS_MSG).ident.span;
+    // `to()` doesn't shorten span, so we shorten it with `until(..)`
+    // and then include it with `to(..)`
+    let span = path.span.until(last_path_span).to(last_path_span);
```

Comment:
```suggestion
    let span = path.span.with_hi(last_path_span.hi());
```
should work here. And it is a little bit clearer what is done here IMO.